flink-streaming-java/src/test/java/org/apache/flink/streaming/api/graph/SinkTransformationTranslatorTest.java

```diff
@@ -0,0 +1,253 @@
+import org.apache.flink.api.common.typeutils.base.IntSerializer;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.ExecutionOptions;
+import org.apache.flink.core.io.SimpleVersionedSerializerTypeSerializerProxy;
+import org.apache.flink.streaming.api.datastream.DataStreamSink;
+import org.apache.flink.streaming.api.datastream.DataStreamSource;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.operators.ChainingStrategy;
+import org.apache.flink.streaming.runtime.operators.sink.BatchCommitterOperatorFactory;
+import org.apache.flink.streaming.runtime.operators.sink.BatchGlobalCommitterOperatorFactory;
+import org.apache.flink.streaming.runtime.operators.sink.GlobalStreamingCommitterOperatorFactory;
+import org.apache.flink.streaming.runtime.operators.sink.StatelessWriterOperatorFactory;
+import org.apache.flink.streaming.runtime.operators.sink.StreamingCommitterOperatorFactory;
+import org.apache.flink.streaming.runtime.operators.sink.TestSink;
+import org.apache.flink.util.TestLogger;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/**
+ * Tests for {@link org.apache.flink.streaming.api.transformations.SinkTransformation}.
+ */
+@RunWith(Parameterized.class)
+public class SinkTransformationTranslatorTest extends TestLogger {
+
+	@Parameterized.Parameters(name = "Execution Mode: {0}")
+	public static Collection<Object[]> data() {
+		return Arrays.asList(new Object[][]{
+				{RuntimeExecutionMode.STREAMING, StreamingCommitterOperatorFactory.class, GlobalStreamingCommitterOperatorFactory.class},
+				{RuntimeExecutionMode.BATCH, BatchCommitterOperatorFactory.class, BatchGlobalCommitterOperatorFactory.class}});
+	}
+
+	@Parameterized.Parameter()
+	public RuntimeExecutionMode runtimeExecutionMode;
+
+	@Parameterized.Parameter(1)
+	public Class<?> committerClass;
+
+	@Parameterized.Parameter(2)
+	public Class<?> globalCommitterClass;
+
+	static final String NAME = "FileSink";
+	static final String SLOT_SHARE_GROUP = "FileGroup";
+	static final String UID = "FileUid";
+	static final int PARALLELISM = 2;
+
+	@Test
+	public void generateWriterTopology() {
+		final StreamGraph streamGraph = buildGraph(
+				TestSink.newBuilder().build(), runtimeExecutionMode);
+
+		final StreamNode sourceNode = findNodeNameContains(streamGraph, "Source");
+		final StreamNode writerNode = findNodeNameContains(streamGraph, "Writer");
+
+		assertThat(streamGraph.getStreamNodes().size(), equalTo(2));
+
+		validateTopology(
+				sourceNode,
+				IntSerializer.class,
+				writerNode,
+				"Writer",
+				StatelessWriterOperatorFactory.class,
+				ChainingStrategy.NEVER,
```

Comment:
Shouldn't the Writer normally be chained to the input?

Reply:
Yes. I just want to verify the `DataStreamSink.disableChaining`.

Reply:
Ah, now I saw that hidden in `setSinkProperty()`. I think we should just test the normal behaviour and then add an extra test for the `disableChaining()`.

Reply:
ok